Content.Shared/Projectiles/ProjectileComponent.cs

```diff
@@ -9,50 +9,84 @@ namespace Content.Shared.Projectiles;
 [RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
 public sealed partial class ProjectileComponent : Component
 {
-    [ViewVariables(VVAccess.ReadWrite), DataField("impactEffect", customTypeSerializer:typeof(PrototypeIdSerializer<EntityPrototype>))]
-    public string? ImpactEffect;
+    /// <summary>
+    ///     The effect that appears when a projectile collides with an entity.
+    /// </summary>
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public EntProtoId? ImpactEffect;
 
     /// <summary>
-    /// User that shot this projectile.
+    ///     User that shot this projectile.
     /// </summary>
-    [DataField("shooter"), AutoNetworkedField]
+    [DataField, AutoNetworkedField]
     public EntityUid? Shooter;
 
     /// <summary>
-    /// Weapon used to shoot.
+    ///     Weapon used to shoot.
     /// </summary>
-    [DataField("weapon"), AutoNetworkedField]
+    [DataField, AutoNetworkedField]
     public EntityUid? Weapon;
 
-    [DataField("ignoreShooter"), AutoNetworkedField]
+    /// <summary>
+    ///     The projectile spawns inside the shooter most of the time, this prevents entities from shooting themselves.
+    /// </summary>
+    [DataField, AutoNetworkedField]
     public bool IgnoreShooter = true;
 
-    [DataField("damage", required: true)] [ViewVariables(VVAccess.ReadWrite)]
+    /// <summary>
+    ///     The amount of damage the projectile will do.
+    /// </summary>
+    [DataField(required: true)] [ViewVariables(VVAccess.ReadWrite)]
     public DamageSpecifier Damage = new();
 
-    [DataField("deleteOnCollide")]
+    /// <summary>
+    ///     If the target should be deleted on collision.
+    /// </summary>
+    [DataField]
     public bool DeleteOnCollide = true;
 
-    [DataField("canPenetrate")]
-    public bool CanPenetrate = false;
+    /// <summary>
+    ///     If the projectile should keep the ability to deal damage after colliding.
+    /// </summary>
+    [DataField]
+    public bool DamageAfterCollide = false;
 
-    [DataField("ignoreResistances")]
+    /// <summary>
+    ///     Penetrate the target only if it has the MobStateComponent.
+    /// </summary>
+    /// <remarks>
+    ///     DeleteOnCollide needs to be false for this to work.
+    /// </remarks>
+    [DataField]
+    public bool CanOnlyPenetrateMobs = false;
+
```

Comment:
This shouldn't be on projectilecomponent, use collision layers or absolute worst case another component if you want this behaviour as projectilecomponent doesn't need to become a god component.